options/posix/generic/pthread.cpp

```diff
@@ -1352,6 +1357,70 @@ int pthread_rwlock_wrlock(pthread_rwlock_t *rw) {
 	return 0;
 }
 
+int pthread_rwlock_timedwrlock(pthread_rwlock_t *rw, const struct timespec *__restrict abstime) {
+	return pthread_rwlock_clockwrlock(rw, CLOCK_REALTIME, abstime);
+}
+
+int pthread_rwlock_clockwrlock(pthread_rwlock_t *rw, clockid_t clock, const struct timespec *__restrict abstime) {
+	SCOPE_TRACE();
+
+	if (rw->__mlibc_flags != 0) {
+		mlibc::panicLogger() << "mlibc: pthread_rwlock_t flags were non-zero"
+			<< frg::endlog;
+	}
+
+	// Adjust for the fact that sys_futex_wait accepts a *timeout*, but
+	// pthread_rwlock_timedwrlock accepts an *absolute time*.
+	struct timespec timeout;
+	if (!mlibc::time_relative_to_absolute(clock, abstime, &timeout))
+		return EINVAL;
```

Comment:
Shouldn't we be doing an `absolute_to_relative` and not a `relative_to_absolute` conversion here?